bindings/py/cpp_src/bindings/math/py_Random.cpp

```diff
@@ -63,16 +63,15 @@ namespace nupic_ext {
 
 
         Random.def("sample",
-            [](Random_t& r, py::array_t<nupic::UInt32>& population, py::array_t<nupic::UInt32>& choices)
+            [](Random_t& r, py::array_t<nupic::UInt32>& population, nupic::UInt32 nSelect)
         {
-            if (population.ndim() != 1 || choices.ndim() != 1)
+            if (population.ndim() != 1 )
             {
                 throw std::runtime_error("Number of dimensions must be one.");
             }
 
-            r.sample(get_it(population), (nupic::UInt)population.size(), get_it(choices), (nupic::UInt)choices.size());
-
-            return choices;
+	    std::vector<nupic::UInt32> tmp_pop(get_it(population), get_it(population) + population.size()); //vector from numpy.array
+            return r.sample(tmp_pop, nSelect);
```

Comment:
I'm returning py list, and should return py::array_t , how do I do that?

Reply:
The pybind11 docs have a section on numpy arrays, which is worth a read:
https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html

Or you can look at the SDR bindings which are in the algorithms directory.  Look for `py::array`.  Here is an example from the SDR bindings: (don't ask me what the capsule does)
```
auto capsule = py::capsule(&self, [](void *self) {});
return py::array(self.getSparse().size(), self.getSparse().data(), capsule);
```